src/java.desktop/share/classes/sun/awt/FontConfiguration.java

```diff
@@ -1249,10 +1251,16 @@ public String getFileNameFromPlatformName(String platformName) {
      * search path.
      */
     public String getExtraFontPath() {
+        if (head == null) {
+            throw new RuntimeException("Fontconfig head is null, check your fonts or fonts configuration");
+        }
```

Comment:
should it be unspecified RuntimeException or InternalError?

Reply:
Hi Sergey, when looking through  the java.desktop codebase I see both used quite a lot,  so probably both would be okay.

Reply:
I interpret InternalError as meaning something wrong with the JDK itself, not something in the environment causing
a problem, so RTE would be fine here.

But this NPE is coming from a field (head) that is initialised by code that reads a binary font configuration file.
Are you quite sure you've diagnosed the root of the problem ? Or that this message is really in the right place ?

What does -Dsun.java2d.debugfonts=true reveal ?

Reply:
> I interpret InternalError as meaning something wrong with the JDK itself, not something in the environment causing a problem, so RTE would be fine here.
> 
> But this NPE is coming from a field (head) that is initialised by code that reads a binary font configuration file. Are you quite sure you've diagnosed the root of the problem ? Or that this message is really in the right place ?
> 
> What does -Dsun.java2d.debugfonts=true reveal ?

Hi Phil, when looking  at the java.desktop  codebase ,  both  RuntimeException and InternalError are used quite a lot for  similar cases.

We had this error on a Linux machine with no fonts,  once some fonts were added the error went away.
I'll try to re-run with the property you suggested.